File: Lv.2/12911.py
def solution(n):
    bin_n = str(format(n, 'b'))
    one_count = bin_n.count('1')
    next = n + 1
    bin_next = str(format(next, 'b'))
    
    if '0' in bin_n:
        while (bin_next.count('1') != one_count):
            next += 1
            bin_next = str(format(next, 'b'))
    else:
        bin_next = '10' + bin_n[1:]
    
    return int(bin_next, 2)

File: programmers/Lv.2/12911.js
function solution(n) {
    var answer = 0;
    let bin_n = n.toString(2);
    let one_count = bin_n.split('1').length - 1
    let next = n+1;
    let bin_next = next.toString(2);

    if (bin_n.includes('0')) {
        while (bin_next.split('1').length - 1 != one_count) {
            next += 1
            bin_next= next.toString(2)
        }        
    } else {
        bin_next = '10' + bin_n.substring(1);
    }
    return parseInt(bin_next, 2);
}
